File: languages/django_models/phenomenon.py
from .abstract_element_model import AbstractElementModel
from django.db import models

class Phenomenon(AbstractElementModel):

    # Mechanics
    expression = models.TextField(blank=True, null=True)
    effects = models.TextField(blank=True, null=True)
    duration = models.PositiveIntegerField(blank=True, null=True)
    catalysts = models.ManyToManyField("Object", blank=True, related_name="phenomenon_catalysts")
    empowerments = models.ManyToManyField("Ability", blank=True, related_name="phenomenon_empowerments")

    # World
    mythology = models.TextField(blank=True, null=True)
    system = models.ForeignKey("Phenomenon", on_delete=models.SET_NULL, blank=True, null=True, related_name="phenomenon_system")
    triggers = models.ManyToManyField("Construct", blank=True, related_name="phenomenon_triggers")
    wielders = models.ManyToManyField("Character", blank=True, related_name="phenomenon_wielders")
    environments = models.ManyToManyField("Location", blank=True, related_name="phenomenon_environments")

    def __str__(self):
        return self.name

File: languages/csharp_unity_json_properties/Phenomenon.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Phenomenon : Element
{
    [JsonProperty("expression"), TextAttribute("")]
    public string expression;
    [JsonProperty("effects"), TextAttribute("")]
    public string effects;
    [JsonProperty("duration"), Integer(0)]
    public int duration;
    [JsonProperty("catalysts"), ReferenceAttribute(typeof(Object), true)]
    public string catalysts;
    [JsonProperty("empowerments"), ReferenceAttribute(typeof(Ability), true)]
    public string empowerments;
    [JsonProperty("mythology"), TextAttribute("")]
    public string mythology;
    [JsonProperty("system"), ReferenceAttribute(typeof(Phenomenon))]
    public string system;
    [JsonProperty("triggers"), ReferenceAttribute(typeof(Construct), true)]
    public string triggers;
    [JsonProperty("wielders"), ReferenceAttribute(typeof(Character), true)]
    public string wielders;
    [JsonProperty("environments"), ReferenceAttribute(typeof(Location), true)]
    public string environments;
}
